frontend/src/component/form/TextField.tsx

```diff
@@ -31,6 +31,12 @@ export class TextField extends Component<IProps, IState> {
     this.props.onChange(text);
   };
 
+  updateValue(text: string) {
```

Comment:
Actually, why not add both defaultText and text and make them both optional props? I am guessing, that's how it is being done internally in `<input>`. The key is to make sure the `text` prop is updated as and when the value changes in `InputField`. Without `changeHandler`, it will keep resetting the value.
```
changeHandler(event) {
    this.setState({ value: event.target.value });
}

<TextField text={this.state.value} onChange={this.changeHandler} />
```

Reply:
@rohithbalaji123 I recommend not to add text prop for now. I am trying to avoid magical value update when possible because that's usually a source of bug. The data should flow in one direction without having to guess what's going on.

Reply:
@byliuyang 
Actually, this introduces two sources of truth for the value contained in the input field. By having this API, we are quite literally adding a way to update the state which the prop already does for us. Isn't it the whole purpose of props?

Now, coming back to the two sources of truth issue, consider a case where onChange listener looks something like this,
```
onChangeValue(value: string) {
    if (isInvalid(value))
        return;
    this.setState({ value });
}
```
In the above scenario, the text field will show one thing and the state which holds the value for usage is something else. I feel having the text as a prop is a more natural and changing that will need developers to keep in mind about this intricacy.

Reply:
@rohithbalaji123 I was thinking the value is an internal state of text field. The only way for the outside world to change it is explicitly invoking update API so that we know exactly where we are updating it's value when reading the code.  I was very confused where did update the long link when doing the previous refactor and introduced this bug. Want to avoid this from happening again.

Reply:
@byliuyang , Okay, in that case, the ideal solution would have been to have a method called `updateLongLink` which sets the state of `longLink`. That seems more like a React way.
```
updateLongLink(to: string) {
    this.setState({ longLink: to });
}
```
https://reactjs.org/docs/refs-and-the-dom.html#when-to-use-refs

Anyways, this method still logically works. I am ready to mark it resolved.

Reply:
@rohithbalaji123 Okey. I can change this.

Reply:
@rohithbalaji123 made changes as you suggested as reading the React doc.